src/main/java/org/kiwiproject/dropwizard/poller/metrics/ClientPollerStatistics.java

```diff
@@ -0,0 +1,128 @@
+package org.kiwiproject.dropwizard.poller.metrics;
+
+import org.kiwiproject.dropwizard.poller.ClientPoller;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * Defines contract for collecting basic statistics on {@link ClientPoller ) instances.
+ */
+//TODO: Do we want to enhance this to be stats per host?
+public interface ClientPollerStatistics {
+    /**
+     * Adds a new latency measurement for how long a poll took.
+     */
+    void addPollLatencyMeasurement(long millis);
+
+    /**
+     * The average number of milliseconds it takes per poll.
+     */
+    double averagePollLatencyInMillis();
+
+    /**
+     * Increment the number of attempted polls.
+     */
+    void incrementCount();
+
+    /**
+     * Number of milliseconds since the epoch that the last poll attempt was made.
+     */
+    Optional<Long> lastAttemptTimeInMillis();
+
+    /**
+     * Increment the number of successful polls.
+     */
+    void incrementSuccessCount();
+
+    /**
+     * Increment the number of times polling was skipped.
+     */
+    void incrementSkipCount();
+
+    /**
+     * Number of milliseconds since the epoch that the last poll attempt was skipped.
+     */
+    Optional<Long> lastSkipTimeInMillis();
+
+    /**
+     * Increment the number of failed polls.
+     */
+    void incrementFailureCount(String message);
+
+    /**
+     * Increment the number of failed polls.
+     */
+    void incrementFailureCount(Throwable exception);
+
+    /**
+     * Number of milliseconds since the epoch that the last failure occurred. Intended to be used for time-based
+     * health checking of a client poller.
+     */
+    Optional<Long> lastFailureTimeInMillis();
+
+    /**
+     * Returns a {@link Stream) over recent timestamps of poll failures.
```

Comment:
Review _all_ the links and make sure they are valid. This one ends with a closing parentheses instead of a closing curly brace. (I tend to look at the JavaDoc preview in IntelliJ to make sure the `link` and `code` are all correct.)